File: java/src/phase 1-conditional_thinking_java/Q29ThirdAngleTriangle.java
import java.util.Scanner;

/**
 * Phase 1: Conditional Thinking (If-Else, Boolean Logic)
 * Level 3: Math and Number Logic
 * Question 29: Take two angles of a triangle and compute the third angle.
 * 
 * Mentor Tips (Java 26):
 * - The sum of all angles in a triangle is 180. The third angle is 180 - (a1 + a2). Check if input angles are valid (> 0 and sum < 180).
 */
public class Q29ThirdAngleTriangle {

    public static double solve(double a1, double a2) {
        // TODO: Implement this method to solve the question
        0.0;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Angle 1: "); double a1 = scanner.nextDouble();\n        System.out.print("Angle 2: "); double a2 = scanner.nextDouble();\n        System.out.println("Third angle: " + solve(a1, a2));
        scanner.close();
    }
}

File: python/phase 1-condtional_thinking/q29_third_angle_triangle.py
"""
Phase 1: Conditional Thinking (If-Else, Boolean Logic)
Level 3: Math and Number Logic
Question 29: Take two angles of a triangle and compute the third angle.

Mentor Tips (Python 3.14+):
- The sum of all angles in a triangle is 180. The third angle is 180 - (a1 + a2). Check if input angles are valid (> 0 and sum < 180).
"""

def solve(a1: float, a2: float) -> float:
    # TODO: Implement this method to solve the question
    if a1 <= 0 or a2 <= 0 or (a1 + a2) >= 180:
        return 0.0
    return 180.0 - (a1 + a2)

def main():
    a1 = float(input("Angle 1: "))
    a2 = float(input("Angle 2: "))
    print(f"Third angle: {solve(a1, a2)}")

if __name__ == "__main__":
    main()
